packages/js-moi-providers/src.ts/base-provider.ts

```diff
@@ -1131,6 +1133,45 @@ export class BaseProvider extends AbstractProvider {
         }
     }
 
+    public async getSubscription(event: ProviderEvents): Promise<string> {
+        let params: unknown = [];
+
+        if (typeof event === "string") {
+            params = event;
+        }
+
+        if (typeof event === "object") {
+            if (event.event === 'newTesseractsByAccount') {
+                if (!isValidAddress(event.params.address)) {
+                    ErrorUtils.throwArgumentError("Invalid address provided", "event.params", event.params);
+                }
+
+                params = [event.event, { address: event.params.address }];
+            }
+
+            if (event.event === 'newLogs') {
+                if (!isValidAddress(event.params.address)) {
+                    ErrorUtils.throwArgumentError("Invalid address provided", "event.params.address", event.params.address);
+                }
```

Comment:
The validation logic looks common for both why don't we keep it outside.

Reply:
https://github.com/sarvalabs/js-moi-sdk/pull/91/commits/39b1d8470938b4d35f6f6a179b2869427c8aa4b6